starknet-rpc-test/get_block_hash_and_number.rs

```diff
@@ -1,46 +1,27 @@
 extern crate starknet_rpc_test;
 
 use rstest::rstest;
-use starknet_ff::FieldElement;
+use starknet_core::types::BlockId;
 use starknet_providers::Provider;
-use starknet_rpc_test::constants::{ARGENT_CONTRACT_ADDRESS, MINT_AMOUNT, SIGNER_PRIVATE};
-use starknet_rpc_test::fixtures::madara;
-use starknet_rpc_test::utils::{create_account, AccountActions};
-use starknet_rpc_test::{MadaraClient, Transaction};
+use starknet_rpc_test::fixtures::{madara, ThreadSafeMadaraClient};
 
 #[rstest]
 #[tokio::test]
-async fn work_ok_at_start_and_with_new_blocks(#[future] madara: MadaraClient) -> Result<(), anyhow::Error> {
-    let madara = madara.await;
-    let rpc = madara.get_starknet_client();
-
-    assert_eq!(
-        rpc.block_hash_and_number().await?.block_hash,
-        FieldElement::from_hex_be("0x031ebd02657f940683ae7bddf19716932c56d463fc16662d14031f8635df52ad").unwrap()
-    );
-    assert_eq!(rpc.block_hash_and_number().await?.block_number, 0);
-
-    madara.create_empty_block().await?;
-    assert_eq!(
-        rpc.block_hash_and_number().await?.block_hash,
-        FieldElement::from_hex_be("0x001d68e058e03162e4864ef575445c38deea4fad6b56974ef9012e8429c2e7b9").unwrap()
-    );
-    assert_eq!(rpc.block_hash_and_number().await?.block_number, 1);
-
-    let account = create_account(rpc, SIGNER_PRIVATE, ARGENT_CONTRACT_ADDRESS, true);
-
-    let token_transfer_tx = Transaction::Execution(account.transfer_tokens(
-        FieldElement::from_hex_be(ARGENT_CONTRACT_ADDRESS).expect("Invalid Contract Address"),
-        FieldElement::from_hex_be(MINT_AMOUNT).expect("Invalid Mint Amount"),
-        None,
-    ));
-
-    madara.create_block_with_txs(vec![token_transfer_tx]).await?;
-    assert_eq!(rpc.block_hash_and_number().await?.block_number, 2);
-    assert_eq!(
-        rpc.block_hash_and_number().await?.block_hash,
-        FieldElement::from_hex_be("0x049b84477d7b0e2f6d6e3cf7dffcb8e5e12b6bb07f673daf7e85b06e69fd041b").unwrap()
-    );
+async fn work_ok_at_start_and_with_new_blocks(madara: &ThreadSafeMadaraClient) -> Result<(), anyhow::Error> {
+    let rpc = madara.get_starknet_client().await;
+
+    {
+        let _madara_write_lock = madara.write();
+        let block_number = rpc.block_number().await?;
+        let (hash, number) = match rpc.get_block_with_tx_hashes(BlockId::Number(block_number)).await.unwrap() {
+            starknet_core::types::MaybePendingBlockWithTxHashes::Block(b) => (b.block_hash, b.block_number),
+            _ => panic!(),
+        };
+
+        let res = rpc.block_hash_and_number().await?;
+        assert_eq!(res.block_hash, hash);
+        assert_eq!(res.block_number, number);
```

Comment:
does this really test anything? we are comparing two rpc endpoint results. if they're fetched from the same place internally, it won't catch an error in hash calculation

Reply:
We can't check against a specific block number or block hash because the order of the tests being run is non deterministic.
That's the best we can do here. Make sure that the RPC routes are internally coherent.

Reply:
got it